src/huggingface_hub/_snapshot_download.py

```diff
@@ -214,6 +215,12 @@ def snapshot_download(
                     f"Returning existing local_dir `{local_dir}` as remote repo cannot be accessed in `snapshot_download` ({api_call_error})."
                 )
                 return str(local_dir.resolve())
+            # If local_dir doesn't exist or is empty, raise an error when in offline mode
+            raise LocalEntryNotFoundError(
+                f"Cannot download files to the specified local directory '{local_dir}' as remote repo cannot be "
+                f"accessed in offline mode. Please check your internet connection or set HF_HUB_OFFLINE=0 to enable "
+                f"online mode."
+            )
```

Comment:
If you remove this error, we would still raise an error below at 

```py
raise LocalEntryNotFoundError(
                "An error happened while trying to locate the files on the Hub and we cannot find the appropriate"
                " snapshot folder for the specified revision on the local disk. Please check your internet connection"
                " and try again."
            ) from api_call_error
```

which IMO is more explicit to the user (especially given the `from api_call_error` page)

Reply:
so I'd just remove this change

```suggestion
```